### PR TITLE
Add EIP: Subscription-Based Token

### PR DESCRIPTION
**Pull request by 0xRobinR**
The subscription-based [ERC-20](https://file+.vscode-resource.vscode-cdn.net/p:/Core360/EIP/EIPs/EIPS/eip-20.md) token extends the basic [ERC-20](https://file+.vscode-resource.vscode-cdn.net/p:/Core360/EIP/EIPs/EIPS/eip-20.md) token standard with a subscribe and unsubscribe function, which allow users to subscribe or unsubscribe from the subscription service. The subscriptionFee and subscriptionFrequency variables define the cost and frequency of the subscription. The nextPaymentDate mapping keeps track of the next payment date for each subscriber.

This EIP also proposes adding renewSubscription functions to the token contract, that can be used by token holders to renew their subscription to a service or product that requires recurring payments in the form of the token.

### CODE REVIEW

File: EIPS/eip-360.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+eip: 360
```
Comment (abcoathup) [inline EIPS/eip-360.md:2]:
```suggestion
eip: 6932
```
Assigning EIP number 6932, please also update the filename.
EIP numbers are generally the PR number.  Using 6932 as this was the earliest PR #6932

Comment (0xRobinR) [reply EIPS/eip-360.md:2]: but the current PR is 6933, will that be okay?

Comment (abcoathup) [reply EIPS/eip-360.md:2]: Please use EIP number 6932 as this was the earliest PR.

**Comment by 0xRobinR** [comment]
@abcoathup will you share an update on this ?

**Comment by abcoathup** [comment]
> @abcoathup will you share an update on this ?

You need to add a rationale and security considerations section.  Then you will need an EIP editor to review.

**Comment by 0xRobinR** [comment]
https://github.com/360core/eip-6932

working implementation of eip-6932

@axic @Pandapip1 @SamWilsn @xinbenlv @abcoathup

File: EIPS/eip-6932.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  /// @return get the subscription frequency
+  function subscriptionFrequency() external view returns (uint);
+}
+```
+
+## Rationale
+
+The introduction of the Subscription Token EIP addresses several critical aspects of the Ethereum ecosystem, providing a rationale for its adoption and implementation.
+
+Traditional subscription models often require users to manually initiate recurring payments, leading to inconvenience and potential disruptions in service delivery. By establishing a Subscription Token standard, users can authorize automatic deductions from their balances at predefined intervals. This automation simplifies the subscription management process, ensuring uninterrupted access to subscribed services without the need for repetitive manual actions.
```
Comment (SamWilsn) [inline EIPS/eip-6932.md:97]: This all reads more like it belongs in the Motivation section. The Rationale section should justify technical choices made within the EIP itself, while the Motivation justifies the EIP as a whole.

File: EIPS/eip-6932.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+## Motivation
+
+The rise of subscription-based business models necessitates a standardized approach to handle recurring payments on the Ethereum blockchain. Currently, users often manually initiate subscription payments, resulting in inconvenience and potential disruptions in service delivery. By introducing a Subscription Token, users can seamlessly authorize periodic deductions, enabling uninterrupted access to subscribed services.
+
+The subscription-based [ERC-20](./eip-20.md) token provides a more flexible and convenient way to manage recurring payments. It can be used for a wide range of services and products that require regular payments, such as subscription-based content platforms, gaming services, and more.
+
+## Specification
+
+The Subscription Token implements an interface following the guidelines defined in [ERC-1337](./eip-1337.md), ensuring consistency and interoperability across different implementations. Key features include:
```
[the remark this answers is not in the document]
Comment (0xRobinR) [reply EIPS/eip-6932.md:30]: Sure, I will minimize this

File: EIPS/eip-6932.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+frequency of subscription, interval at which the `subscriptionFee` will be charged.
+
+#### subscribe
+
+function to subscribe toh the subscription
+
+#### unsubscribe
+
+revoke subscription
```
[the remark this answers is not in the document]
Comment (0xRobinR) [reply EIPS/eip-6932.md:64]: I will add more details to this

**Comment by SamWilsn** [comment]
Just checking if this is ready for re-review?

**Comment by 0xRobinR** [comment]
> Just checking if this is ready for re-review?

Yes, just updating with the required changes, I'll push and tag you then

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.

**Pull request closed without merging**